### PR TITLE
Update get vector ut

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

File: tests/ut/test_get_vector.cc [hunk 1]
```diff
@@ -17,25 +17,81 @@
...
     int64_t nb = 10000;
     int64_t dim = 128;
     int64_t seed = 42;
 
-    auto base_gen = [&]() {
+    auto base_bin_gen = [&]() {
         knowhere::Json json;
         json[knowhere::meta::DIM] = dim;
-        json[knowhere::meta::METRIC_TYPE] = knowhere::metric::L2;
+        json[knowhere::meta::METRIC_TYPE] = knowhere::metric::HAMMING;
```
Comment (PwzXxm) [inline tests/ut/test_get_vector.cc:30]: how about other binary metrics?

File: tests/ut/test_get_vector.cc [hunk 2]
```diff
@@ -17,25 +17,81 @@
...
+}
+
+TEST_CASE("Test Float Get Vector By Ids", "[Float GetVectorByIds]") {
+    using Catch::Approx;
+
+    int64_t nb = 10000;
+    int64_t dim = 128;
+    int64_t seed = 42;
+
+    auto metric = GENERATE(as<std::string>{}, knowhere::metric::L2, knowhere::metric::COSINE);
```
Comment (PwzXxm) [inline tests/ut/test_get_vector.cc:89]: how about IP?

File: tests/ut/test_get_vector.cc (hunk 1 above)
Comment (cydrain) [reply tests/ut/test_get_vector.cc:30]: for binary index types, metric type does not matter

File: tests/ut/test_get_vector.cc (hunk 2 above)
Comment (cydrain) [reply tests/ut/test_get_vector.cc:89]: for float index, when COSINE metric type, get vector may return not-original vector.
so it's enough to only test L2 and COSINE

**Pull request merged**